### ISSUE TITLE
MAMEHub 4.0 tries to run csume64.exe on 32bit Win7.

### ISSUE THREAD

**Issue by Xorcist77** [MisterTea/MAMEHub#377]
Seems that MAMEHub is always trying to run csume64.exe instead of csume.exe under 32bit Win7.

The error generated is:

Cannot run program "csume64.exe": CreateProcess error=193, %1 is not a valid Win32 application
java.lang.ProcessBuilder.start(Unknown Source)
java.lang.Runtime.exec(Unknown Source)
java.lang.Runtime.exec(Unknown Source)
com.mamehub.client.MameHubEngine.launchGame(MameHubEngine.java:152)
com.mamehub.client.MainFrame$4.mouseClicked(MainFrame.java:617)
java.awt.AWTEventMulticaster.mouseClicked(Unknown Source)
java.awt.Component.processMouseEvent(Unknown Source)
javax.swing.JComponent.processMouseEvent(Unknown Source)
java.awt.Component.processEvent(Unknown Source)
java.awt.Container.processEvent(Unknown Source)
java.awt.Component.dispatchEventImpl(Unknown Source)
java.awt.Container.dispatchEventImpl(Unknown Source)
java.awt.Component.dispatchEvent(Unknown Source)
java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
java.awt.Container.dispatchEventImpl(Unknown Source)
java.awt.Window.dispatchEventImpl(Unknown Source)
java.awt.Component.dispatchEvent(Unknown Source)
java.awt.EventQueue.dispatchEventImpl(Unknown Source)
java.awt.EventQueue.access$200(Unknown Source)
java.awt.EventQueue$3.run(Unknown Source)
java.awt.EventQueue$3.run(Unknown Source)
java.security.AccessController.doPrivileged(Native Method)
java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
java.awt.EventQueue$4.run(Unknown Source)
java.awt.EventQueue$4.run(Unknown Source)
java.security.AccessController.doPrivileged(Native Method)
java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
java.awt.EventQueue.dispatchEvent(Unknown Source)
java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
java.awt.EventDispatchThread.pumpEvents(Unknown Source)
java.awt.EventDispatchThread.pumpEvents(Unknown Source)
java.awt.EventDispatchThread.run(Unknown Source)

**Comment by Gitweazle** [comment]
Make sure you delete or rename csume64.exe because MAMEHub prioritizes it over the 32-bit executable csume.exe

**Comment by Xorcist77** [comment]
Yes that worked, I renamed the csume64.exe so it would find the csume.exe.

Does Java not supply a reliable means of determining the underlying processor architecture? Seems like there should be some way to determine if it's x86 or x64 and use the proper csume rather than potentially incorrect defaulted prioritization.
